runtime/scripts/math.js

```diff
@@ -506,6 +506,26 @@ var math = Numbas.math = /** @lends Numbas.math */ {
         return Math.abs(a-b) <= Math.max( rel_tol * Math.max(Math.abs(a), Math.abs(b)), abs_tol );
     },
 
+    is_scalar_multiple(a,b,rel_tol,abs_tol) {
+      rel_tol = rel_tol === undefined ? 1e-15 : rel_tol;
+      abs_tol = abs_tol === undefined ? 1e-15 : abs_tol;
+
+      let ratios = [];
+
+      a.forEach(function (aElement, index) {
+          const bElement = b[index];
+          if (is_close(aElement, 0) || is_close(bElement, 0)) {
+              if (!is_close(aElement, bElement)) {
```

Comment:
Should this be `math.isclose(bElement, 0)`? "Closeness" isn't a transitive property, so it's possible that `a` is close to `0`, and `b` is close to `a`, but `b` is not close to `0`.